Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -16,212 +15,133 @@
-            success = False
-            for i in range(0, 5):
-                time.sleep(5)
-                if verify_hybrid_worker() > 0:
-                    success = True
-                    break
-            if not success:
-                log(ERROR, "Linux Hybrid worker process was not detected")
-                return [-1]
-        except SystemExit:
-            # very important for the first child of the start_daemon method to die properly
-            exit(0)
-        except Exception, exception:
-            log(ERROR, exception.message)
-            return [-1]
-        log(INFO, "Set method exited successfully for Enabled = True")
-    else:
-        # enabled is set to false
-        try:
-            kill_hybrid_worker(WorkspaceId)
-            if os.path.isfile(WORKER_CONF_FILE_PATH):
-                os.remove(WORKER_CONF_FILE_PATH)
-            if os.path.isfile(WORKER_STATE_FILE_PATH):
-                os.remove(WORKER_STATE_FILE_PATH)
-        except Exception, exception:
-            log(ERROR, exception.message)
-            return [-1]
-        log(INFO, "Set method exited successfully for Enabled = False")
-    return [0]
 
+def Test_Marshall(ResourceSettings):
+    """
+    Test method for the DSC resoruce
+    If it returns [0] no further action is taken
+    If it returns [-1] Set_Marshall is called
+    It tests for whether oms.conf exists and the worker manager is running and the latest version
+    :param ResourceSettings:
+    :return: [0] if all tests pass [-1] otherwise
+    """
+    workspace_id, azure_dns_agent_svc_zone, updates_enabled, diy_enabled = read_settings_from_mof_json(ResourceSettings)
+    if not is_oms_primary_workspace(workspace_id):
+        # not primar workspace
+        # return unconditional [0] for a NOOP on non-primary workspace
+        log(DEBUG, "Test_Marshall skipped: non primary workspace. Test_Marshall returned [0]")
+        return [0]
 
-def Test_Marshall(WorkspaceId, Enabled, AzureDnsAgentSvcZone):
-    init_locals(WorkspaceId, AzureDnsAgentSvcZone)
-    try:
-        isprimary = is_oms_primary_workspace(WorkspaceId)
-    except Exception, e:
-        log(ERROR, "Could not detect if workspace is primary\n %s" % (str(e)))
+    if not os.path.isfile(OMS_CONF_FILE_PATH):
+        log(INFO, "Test_Marshall returned [-1]: oms.conf file not found")
+        return [-1]
+    if (updates_enabled or diy_enabled) and not is_worker_manager_running_latest_version(workspace_id):
+        # Either the worker manager is not running, or its not latest
+        log(INFO, "Test_Marshall returned [-1]: worker manager isn't running or is not latest")
+        return [-1]
+    if not updates_enabled and not diy_enabled and get_worker_manager_pid_and_version(workspace_id, False)[0] > 0:
+        log(INFO, "Test_Marshall returned [-1]: worker manager is running when no solution is enabled")
+        return [-1]
+    if not is_oms_config_consistent_with_mof(updates_enabled, diy_enabled):
```

Comment:
if is_oms_config_consistent_with_mof(updates_enabled, diy_enabled) is False? More readable?